normandy/recipes/utils.py

```diff
@@ -115,3 +116,33 @@ def sign_data(self, content_list):
                 'public_key': res['public_key'],
             })
         return signatures
+
+
+def verify_signature(data, signature, pubkey):
+    if isinstance(data, str):
+        data = data.encode()
+
+    # add data template
+    data = b'Content-Signature:\x00' + data
+
+    signature = base64.b64decode(un_urlsafe(signature))
+    pubkey = un_urlsafe(pubkey)
+
+    verifying_pubkey = ecdsa.VerifyingKey.from_pem(pubkey)
+    verified = verifying_pubkey.verify(
+        signature,
+        data,
+        hashfunc=hashlib.sha384,
+        sigdecode=ecdsa.util.sigdecode_string)
+
+    print(dir(verifying_pubkey))
+
+    return verified
+
+
+def un_urlsafe(input):
+    input = input.replace("_", "/")
+    input = input.replace("-", "+")
+    if len(input) % 4 > 0:
+        input += "=" * (4 - len(input) % 4)
```

Comment:
Autograph is weird. It uses this not-quite-base64 thing that uses `_-` instead of `/+`, and doesn't include the padding, which we need for `base64.b64decode`. Maybe `base64.urlsafe_b64decode` will deal with all this for us, I'll give it shot. Otherwise I'll add a docstring to this function.